openoa/analysis/yaw_misalignment.py

```diff
@@ -0,0 +1,571 @@
+            self.yaw_misalignment_std = np.empty([len(self.turbine_ids)])
+            self.yaw_misalignment_95ci = np.empty([len(self.turbine_ids), 2])
+            self.yaw_misalignment_avg_ws = np.empty([len(self.turbine_ids), len(self.ws_bins)])
+            self.yaw_misalignment_std_ws = np.empty([len(self.turbine_ids), len(self.ws_bins)])
+            self.yaw_misalignment_95ci_ws = np.empty([len(self.turbine_ids), len(self.ws_bins), 2])
+
+        elif not self.UQ:
+            inputs = {
+                "power_bin_mad_thresh": [self._power_bin_mad_thresh],
+                "max_power_filter": [self._max_power_filter],
+            }
+            self.inputs = pd.DataFrame(inputs)
+
+            # For saving power or power coefficient as a function of wind vane for each wind speed bin
+            self.power_values_vane_ws = np.empty(
+                [len(self.turbine_ids), len(self.ws_bins), len(self._vane_bins)]
+            )
+
+            # For saving cosine curve fit parameters, yaw misalignment, and mean wind vane angle for each wind speed bin
+            self._curve_fit_params_ws = np.empty([len(self.turbine_ids), len(self.ws_bins), 3])
+            self.yaw_misalignment_ws = np.empty([len(self.turbine_ids), len(self.ws_bins)])
+            self.mean_vane_angle_ws = np.empty([len(self.turbine_ids), len(self.ws_bins)])
+
+            # For saving yaw misalignment and mean wind vane angle averaged over all wind speed bins
+            self.yaw_misalignment = np.empty([len(self.turbine_ids)])
+            self.mean_vane_angle = np.empty([len(self.turbine_ids)])
+
+            self.num_sim = 1
+
+    @logged_method_call
+    def _remove_power_curve_outliers(self, turbine_id):
+        """
+        Removes power curve outliers for a specific turbine by removing timestamps where the pitch
+        angle is above a threshold and timestamps where the wind speed is more than a specific
+        threshold from the median wind speed in each power bin. The filtered turbine data frame is
+        meant to include timestamps when the turbine is operating normally in below-rated
+        conditions.
+
+        Args:
+            turbine_id (str): The name of the turbine for which power curve outlier removal will be performed.
+        """
+
+        # Limit to pitch angles below the specified threshold
+        self._df_turb = self._df_turb.loc[self._df_turb["WROT_BlPthAngVal"] <= self._pitch_thresh]
+
+        # Apply bin-based filter to flag samples for which wind speed is greater than a threshold from the median
+        # wind speed in each power bin
+        turb_capac = self.plant.asset.loc[turbine_id, "rated_power"]
+        bin_width_frac = 0.04 * (self._run.max_power_filter - 0.01)
+        flag_bin = filters.bin_filter(
+            bin_col=self._df_turb["WTUR_W"],
+            value_col=self._df_turb["WMET_HorWdSpd"],
+            bin_width=bin_width_frac * turb_capac,
```

Comment:
I think it makes sense for them to be customizable. I added two new arguments to the `run` method so those two filtering parameters can be specified: `num_power_bins` and `min_power_filter`.

Reply:
I'm not seeing this reflected, do you need to push the changes still?

Reply:
Hi Rob, I checked and those changes are in the code. In addition to the arguments to the `run` method I mentioned above, the `bin_min` argument to the bin filter function uses the `self._min_power_filter` value instead of the hardcoded value of 0.01 used previously. And the `bin_width_frac` variable used in the `bin_width` argument is based on `self._num_power_bins`, which is an argument to `run` instead of hardcoded to 25. Let me know if I'm missing something, or perhaps I misunderstood your original comment.

Reply:
Ok, I think I didn't read closely enough, and looked for the wrong change. This is great, thank you!